### PR TITLE
[cors] add OPTIONS status code + fix function typo

### PR DESCRIPTION
**Pull request by commit-master**
I believe we should handle the StatusCode when receiving OPTIONS requests. Best practice is to respond with a `200` or `204` code.

`200` code seems to be the most used since it seems to work across any browser:
https://stackoverflow.com/questions/46026409/what-are-proper-status-codes-for-cors-preflight-requests

We should let the user override this status code this is why I exposed the `OptionStatusCode` method that take an int.

### CODE REVIEW

File: cors.go [hunk 1]
```diff
@@ -251,7 +255,17 @@ func AllowedOriginValidator(fn OriginValidator) CORSOption {
 	}
 }
 
-// ExposeHeaders can be used to specify headers that are available
+// OptionStatusCode sets a custom status code to use to respond to OPTIONS requests
+// by default it is set to 200 to ensure max compatibility, see:
+// https://stackoverflow.com/questions/46026409/what-are-proper-status-codes-for-cors-preflight-requests
+func OptionStatusCode(code int) CORSOption {
```
Comment (elithrar) [inline cors.go]: I would rephrase this:

- When a handler returns and the status code has not been set prior to writing the body, Go automatically sets the status as HTTP 200. We should make it clear that you do not have to set this option unless you want non-default behaviour - i.e. HTTP 204.
- Refer to the RFC (in this case, the Fetch spec), not StackOverflow.

Comment (commit-master) [reply cors.go]: Thanks for your feedback and your time !

For a bit more context, here is our use case:
our use case

We need the CORS module to be able to set a status code on the OPTION requests so we do not add noise to our monitoring systems with 0 status codes all over the place. Indeed the client is served a 200 thanks to Go's default behaviour, but I believe this could be helpful to anyone that needs monitoring.

I will update the comment to reflect this better.

Comment (elithrar) [reply cors.go]: OK, this makes more sense.

I would suggest that your monitoring middleware wraps the ResponseWriter and captures the final status code (and response duration too), rather than relying on what other middleware are doing.

File: cors.go [hunk 2]
```diff
@@ -251,7 +255,21 @@ func AllowedOriginValidator(fn OriginValidator) CORSOption {
 	}
 }
 
-// ExposeHeaders can be used to specify headers that are available
+// OptionStatusCode sets a custom status code on the OPTIONS requests.
+// Default behaviour sets it to 200 to reflect best practices. This is option is not mandatory
+// and can be used if you need a custom status code (i.e 204) or have a monitoring
```
Comment (elithrar) [inline cors.go]: Remove the piece about monitoring middleware - as per my other comment, there are more robust ways to get the real status code in middleware when logging.

File: cors.go (hunk 1 above)
Comment (commit-master) [reply cors.go]: We already wrap the `ResponseWriter` in our monitoring middleware, but we can't use it in the case of an `OPTIONS` request since we do not perform any call on a backend (thus we never get a response).

File: cors.go (hunk 2 above)
Comment (commit-master) [reply cors.go]: Totally agree we should keep this generic. I'm removing it.

**Comment by elithrar** [comment]
Thanks @commit-master!
